### ISSUE TITLE
Unable to resolve multiple matches for function reports single match

### ISSUE THREAD

**Issue by nijel** [michaeljones/breathe#194]
While building Gammu docs I currently get tons of following errors:

```
WARNING: doxygenfunction: Unable to resolve multiple matches for function "GSM_GetModelInfo" with arguments () in doxygen xml output for project "project0" from directory: .../gammu-doc/xml.
Potential matches:
    - GSM_PhoneModel * GSM_GetModelInfo(GSM_StateMachine *)
```

The error message really does not help - it says multiple matches, but shows only one. Maybe I'm just doing something wrong...

It used to work in some previous breathe versions, though I don't know when it has started to fail (it's probably long time ago without anybody noticing).

**Comment by vitaut** [comment]
What version of Breathe do you use? The following works fine on master:

```c++
/** GSM_PhoneModel */
class GSM_PhoneModel;

/** GSM_StateMachine */
class GSM_StateMachine;

/** GSM_GetModelInfo */
GSM_PhoneModel * GSM_GetModelInfo(GSM_StateMachine *);
```

```
.. doxygenfunction:: GSM_GetModelInfo
```

**Comment by nijel** [comment]
It's 3.2.0, with 4.0.0 I get following exception:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/cmdline.py", line 254, in main
    app.build(force_all, filenames)
  File "/usr/lib/python2.7/dist-packages/sphinx/application.py", line 220, in build
    self.builder.build_update()
  File "/usr/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 214, in build_update
    'out of date' % len(to_build))
  File "/usr/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 234, in build
    purple, length):
  File "/usr/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 134, in status_iterator
    for item in iterable:
  File "/usr/lib/python2.7/dist-packages/sphinx/environment.py", line 478, in update_generator
    self.read_doc(docname, app=app)
  File "/usr/lib/python2.7/dist-packages/sphinx/environment.py", line 628, in read_doc
    pub.publish()
  File "/usr/lib/python2.7/dist-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "/usr/lib/python2.7/dist-packages/docutils/readers/__init__.py", line 72, in read
    self.parse()
  File "/usr/lib/python2.7/dist-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/__init__.py", line 172, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 171, in run
    input_source=document['source'])
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2727, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 328, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 396, in new_subsection
    node=section_node, match_titles=True)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 283, in nested_parse
    node=node, match_titles=match_titles)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2302, in explicit_markup
    self.explicit_list(blank_finish)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2332, in explicit_list
    match_titles=self.state_machine.match_titles)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 320, in nested_list_parse
    node=node, match_titles=match_titles)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python2.7/dist-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2605, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2312, in explicit_construct
    return method(self, expmatch)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2055, in directive
    directive_class, match, type_name, option_presets)
  File "/usr/lib/python2.7/dist-packages/docutils/parsers/rst/states.py", line 2104, in run_directive
    result = directive_instance.run()
  File "/usr/lib/python2.7/dist-packages/breathe/directives.py", line 501, in run
    mask_factory)
  File "/usr/lib/python2.7/dist-packages/breathe/directive/base.py", line 88, in render
    return object_renderer.render()
  File "/usr/lib/python2.7/dist-packages/breathe/renderer/rst/doxygen/compound.py", line 220, in render
    self.update_signature(signode)
  File "/usr/lib/python2.7/dist-packages/breathe/renderer/rst/doxygen/compound.py", line 333, in update_signature
    signode.children[0][0] = self.node_factory.Text("enum ")
  File "/usr/lib/python2.7/dist-packages/docutils/nodes.py", line 582, in __setitem__
    self.children[key] = item
IndexError: list assignment index out of range
```

**Comment by michaeljones** [comment]
Hi @nijel,

Are the Gammu docs openly accessible? Is this something we can try to reproduce and debug on our side? It is kind of you to include the stack trace but in this case we'd benefit from a bit more information. Ideally we'd get you on to the latest version of breathe as we don't really have the time to bug fix older versions. We're only a small project :)

Cheers for reporting the issue though and for following up. It definitely seems like there are some things to fix here.

Michael

**Comment by nijel** [comment]
Yes, but let's keep the exception as separate issue, see #195.

**Comment by nijel** [comment]
Tested with current master branch and I still get the same error - it complains about multiple matches, but prints just one:

```
gammu/docs/manual/c/info.rst:13: WARNING: doxygenfunction: Unable to resolve multiple matches for function "GSM_GetFirmware" with arguments () in doxygen xml output for project "project0" from directory: .../gammu-doc/xml.
Potential matches:
    - GSM_Error GSM_GetFirmware(GSM_StateMachine  *, char *, char *, double *)
```

**Comment by michaeljones** [comment]
Thanks for trying it @nijel. Sorry I didn't follow through with that. You're right that it shouldn't be doing that I don't think. Hopefully it should be an easy fix.

**Comment by nijel** [comment]
With current master I can no longer reproduce this issue, so #205 seems to worked well.

**Comment by melvinvermeeren** [comment]
Based upon comments above I assume fixed by #205 since 2015.

Assigned to @michaeljones since he is the one that fixed the issue.

Closing for issue tracker cleanup.
